src/backend/joanie/payment/backends/dummy.py

```diff
@@ -316,3 +317,36 @@ def tokenize_card(self, order=None, billing_address=None, user=None):  # pylint:
             "customer": str(user.id),
             "card_token": f"card_{user.id}",
         }
+
+    def cancel_or_refund(self, amount, transaction_reference):
+        """
+        Dummy method to refund an installment by taking the transction reference (`payment_id`
+        in the cache). This method only treats a refund.
+        """
+        resource = cache.get(transaction_reference)
+        if not resource:
+            raise exceptions.RegisterPaymentFailed(
+                f"Resource {transaction_reference} does not exist, cannot refund."
+            )
+
+        transaction = Transaction.objects.get(reference=transaction_reference)
+        if transaction.total != amount:
+            raise exceptions.RefundPaymentFailed(
+                f"Resource {transaction_reference} amount does not match the amount to refund"
+            )
+
+        # Trigger post notification for Dummy usage
+        notification_request = APIRequestFactory().post(
+            reverse("payment_webhook"),
+            data={
+                "id": transaction.reference,
+                "type": "refund",
+                "state": "success",
+            },
+            format="json",
+        )
+        notification_request.data = json.loads(
+            notification_request.body.decode("utf-8")
+        )
+
+        return True
```

Comment:
Why returning a boolean ?

Reply:
I will return a dictionary with this data : 
```
{
   "id": transaction.reference,
   "type": "refund",
   "state": "success",
 }
```